src/plugins/data/common/index_patterns/index_patterns/types.ts

```diff
@@ -33,3 +33,16 @@ export interface TypeMeta {
   aggs?: Record<string, AggregationRestrictions>;
   [key: string]: any;
 }
+
+export interface IndexPatternSpec {
+  id: string;
+  type: string;
+  source: {
+    title: string;
+    timeFieldName: string;
+    sourceFilters: string;
+    fields: string;
+    fieldFormatMap: string;
```

Comment:
i wonder if sufficient information to rebuild this one can be inside the FieldSpec ? (taking a deeper look)

Reply:
confirmed that indexPattern.fields[x].format === indexPattern.fieldFormatMap[y] where indexPattern.fields[x].name === y.
so fieldFormatMap should be restored from the FieldSpec[]

Reply:
I'd rather not change this now. I have other PRs on the back burner that change this relationship so there's less dependency passing.

Reply:
you don't need to change any logic on the index pattern, just make toSpec omit fieldFormatMap property and when creating index pattern from spec you do something like:

```
 this.fields.forEach(f => this.fieldFormatMap[f.name] = f.format);
```

Reply:
Relevant links to where I'd like to take this 

https://github.com/elastic/kibana/pull/67817/files#diff-fb941bf99aa860c63cd43baf6b56ef33R161
https://github.com/elastic/kibana/pull/67817/files#diff-bc3c1509c7d6e69f09578a2906da185fR381

Reply:
this doesn't conflict with that, that function can still work if fieldFormatMap is internal index pattern structure to help it do things like this, but spec is state that we willl share to the outside and should be minimal and not contain any internal structures.

fields do have a format defined (they are called field formats for a reason :)) if we want to make Field simpler and not expose getFieldFormat function that's fine, but the field format configuraton (FieldFormatSpec) should still be defined on it.

as we want to make index pattern class static we probably don't want that function on the index pattern class either (but on the index patterns service, or even better, talk directly to field formats service if you want to get an instance out of spec) but thats out of scope of this PR.